### ISSUE TITLE
screen size out of sync if host terminal resized while reading from stdin

### ISSUE THREAD

**Issue by allisonkarlitskaya** [neovim/neovim#18915]
### Neovim version (nvim -v)

NVIM v0.7.0

### Vim (not Nvim) behaves the same?

No.  VIM - Vi IMproved 8.2 does not have this issue.

### Operating system/version

Fedora 36

### Terminal name/version

vte (gnome-terminal, others)

### $TERM environment variable

xterm-256color

### Installation

Official repository neovim-0.7.0-1.fc36.x86_64

### How to reproduce the issue

1. Open a terminal window, make it small.
2. Run `sleep 5 | nvim -`
3. Maximize the terminal window.

### Expected behavior

After 5 seconds nvim should be done reading the buffer and should fill the screen.

### Actual behavior

After 5 seconds nvim is done reading the buffer, but it fills the old size of the terminal window.

See the screenshot.

![Screenshot from 2022-06-09 22-35-22](https://user-images.githubusercontent.com/36541154/172940363-4c26e065-393f-40bb-83f4-a0b08d941e13.png)

Any further attempts to resize the window will return things to normal.

This happens equally with `nvim -u NORC -`.

It's worth noting that this *doesn't* happen with vim, because vim waits until it reads the entire buffer before initialising the terminal.

Also note: if you merely resize the window instead of maximizing it, then gnome-terminal will shrink back down to the original size once reading stdin is done.  Maximizing the window prevents the size-change, though, and you're left with the out-of-sync size, as shown in the screenshot.

**Comment by allisonkarlitskaya** [comment]
Interestingly, in another container, I have NVIM v0.6.1 installed and the behaviour with that version is correct when the window is maximized (ie: the content fills the window correctly) but for resized windows, the window will still be forced back to the old size.

So this issue got worse somewhere between 0.6.1 and 0.7.0.

**Comment by muniter** [comment]
Looks like a duplicate of https://github.com/neovim/neovim/issues/15044
